.github/workflows/release-basetools.yml

```diff
@@ -34,12 +34,12 @@ jobs:
             output_dir: "BaseTools\\Bin\\Win64"
             target: AARCH64
 
-          - os: ubuntu-20.04
+          - os: ubuntu-24.04

```

Comment:
@apop5 I actually am concerned, now that I look at it more, that this is going to cause runtime issues for people running on any ubuntu version lower than 24.04. Can you test that by using a 22.04 WSL image with these basetools?

Reply:
One unfortunate thing about us squashing commits and cleaning up when rebasing off of edk2, is we lose the git history - But see this PR for us going down to 20.04 for the reason I described above. https://github.com/microsoft/mu_basecore/pull/467. It is unfortunately not very descriptive, but @kuqin12 am I correct in remembering that we had to do this because using an older version of ubuntu then what it is compiled with results in dll loading errors or something like that?

Reply:
Yes, updating to 24.04 will pull in a later version of libc, which will not work for folks that are using older version of ubuntu.

Reply:
@kuqin12 @Javagedes 

the ubuntu 20.04 runner has been depreciated on github

https://github.com/actions/runner-images/issues/11101


Attempting to use the runner will result in a failed/canceled job, as experienced on: https://github.com/microsoft/mu_basecore/pull/1342

If libc is a problem with 24.04, would limiting the update to Ubuntu 22.04 help this at all?

Reply:
Can't we just build and publish both? They're both valid and we're the builder and publisher. I didn't really care too much in my review because we control most of the environments that would use this and since 24.04 is available and LTS, I chose to go with not dealing with this again as soon in the future if we choose one. But we don't have to.

Reply:
There is no need to compile both because lower versions work with greater versions. But because GitHub is dropping support for 20.04 we are basically being forced to also drop support for 20.04 pre-compiled base tools. Unless we manually compile and publish or have a separate self-maintained 22.04 container.

I think incrementing to 22.04 is the right thing to do, with some kind of acknowledgment that if someone wants to compile on 20.04, they must compile their own base tools locally instead of using the extdep. This is already what happens if someone wants to compile with 18.04 for some reason

Reply:
My statement about compiling for both is not about technical compatibility. It is about making it simple to track and deprecate individual builds in the future. If there was a 20.04 build, it would simply be removed from the workflow with the VM image change and anyone using the latest 20.04 binaries would keep working because their "latest" stays at that 20.04 based release. Anyone that wanted to move to a newer OS image would have an easy and obvious choice as to which to use because they would simply match their Ubuntu image 1:1 to the version in published feed.

Reply:
I'm obviously fine with 22.04 too to solve the immediate problem. But I don't think that's as sustainable as building and tracking releases per Ubuntu version which also allows future hot patches to easily be picked up for the given Ubuntu version, etc.

Reply:
I'll open a GitHub issue to track  having specific versions based on Ubuntu releases, and switching to this for immediate resolution.

Reply:
https://github.com/microsoft/mu_basecore/issues/1346

Reply:
> I'm obviously fine with 22.04 too to solve the immediate problem. But I don't think that's as sustainable as building and tracking releases per Ubuntu version which also allows future hot patches to easily be picked up for the given Ubuntu version, etc.

Changing this to 22.04 is fine. But I wonder if we can get rid of the glibc reliance provided by the system. Otherwise we are bound to _some_ version regardless. I used to think this will get resolved by move base tools to python. But no one seems to like it. Should we resurrect the effort..?